tests/test_verifyStats.py

```diff
@@ -0,0 +1,264 @@
+        config.isTrimmed = True
+        config.rngSeed = 12345
+        darkExposure = isrMock.DarkMock(config=config).run()
+
+        config.rngSeed = 54321
+        fakeDark = isrMock.DarkMock(config=config).run()
+
+        self.inputExp = darkExposure.clone()
+        mi = self.inputExp.getMaskedImage()
+        mi.scaledMinus(1.0, fakeDark.getMaskedImage())
+        makeMockExp(self.inputExp)
+
+        self.camera = isrMock.IsrMock().getCamera()
+
+    def test_dark(self):
+        # Test a subset of the output values to identify that the
+        # image stat methods haven't changed.
+        task = cpVerify.CpVerifyDarkTask()
+        results = task.run(self.inputExp, self.camera)
+        resultStats = results.outputStats
+
+        self.assertAlmostEqual(resultStats['AMP']['C:0,0']['MEAN'], 2.0043, 4)
+        self.assertAlmostEqual(resultStats['AMP']['C:0,0']['NOISE'], 3.12948, 4)
+        self.assertAlmostEqual(resultStats['AMP']['C:0,0']['CR_NOISE'], 3.15946, 4)
+
+        self.assertIn(resultStats['SUCCESS'], [True, False])
+
+
+class VerifyDefectsTestCase(lsst.utils.tests.TestCase):
+    """Unit test for stats code.
+    """
+
+    def setUp(self):
+        # Generate a mock exposure/camera to test.
+        config = isrMock.IsrMockConfig()
+        config.isTrimmed = True
+        config.doGenerateImage = True
+        config.doAddFringe = False
+        config.doAddSource = False
+        config.doAddSky = True
+        config.doAddOverscan = False
+        config.doAddCrosstalk = False
+        config.doAddBias = False
+        config.doAddDark = False
+        config.doAddFlat = False
+        config.doAddFringe = False
+
+        config.skyLevel = 1000
+        config.rngSeed = 12345
+        self.inputExp = isrMock.IsrMock(config=config).run()
+
+        # These are simulated defects
+        self.inputExp.getImage().getArray()[0, 0] = -100000
+        self.inputExp.getImage().getArray()[40, 50] = 100000
+        self.inputExp.getImage().getArray()[75, 50] = np.nan
+
+        makeMockExp(self.inputExp, addCR=False)
+
+        self.inputExp.getMask().getArray()[0, 0] = 1
+        self.inputExp.getMask().getArray()[40, 50] = 1
+        self.inputExp.getMask().getArray()[75, 50] = 1
+
+        self.inputExp.writeFits("/home/czw/tmp/fake_flat.fits")
+        self.camera = isrMock.IsrMock().getCamera()
+
+    def test_defects(self):
+        # Test a subset of the output values to identify that the
+        # image stat methods haven't changed.
+        task = cpVerify.CpVerifyDefectsTask()
+        results = task.run(self.inputExp, self.camera)
+        resultStats = results.outputStats
+
+        self.assertEqual(resultStats['AMP']['C:0,0']['DEFECT_PIXELS'], 53)
+        self.assertEqual(resultStats['AMP']['C:0,0']['OUTLIERS'], 17)
+        self.assertEqual(resultStats['AMP']['C:0,0']['STAT_OUTLIERS'], 3)
+        self.assertAlmostEqual(resultStats['AMP']['C:0,0']['MEDIAN'], 999.466, 4)
```

Comment:
Seems odd to require equivalence to 4 decimal places when only 3 are specified (I guess 0s are assumed!)

Reply:
Expanded value is `999.4660034179688`.